specreduce/tracing.py

```diff
@@ -177,24 +175,31 @@ class KosmosTrace(Trace):
         guess position. Useful for tracing faint sources if multiple
         traces are present, but potentially bad if the trace is
         substantially bent or warped. [default: None]
+    trace_model : `~astropy.modeling.Model` or, optional
+        The 1-D polynomial model used to fit the trace to the bins' peak
+        pixels. Valid options are ``Spline1D``, ``Legendre1D``,
+        ``Chebyshev1D``, and ``Polynomial1D``. ``Spline1D`` is used with
+        the ``SplineSmoothingFilter`` fitter; the other models are
+        paired with the ``LevMarLSQFitter`` fitter.
+        [default: ``models.Spline1D(degree=3)``]
     peak_method : string, optional
-        One of ``gaussian`` (default), ``centroid``, or ``max``.
-        gaussian: fits a gaussian to the window within each bin and
-        adopts the central value.  centroid: takes the centroid of the
-        window within in bin.  smooth_max: takes the position with the
-        maximum flux after smoothing over the window within each bin.
-
-    Improvements Needed
-    -------------------
-    1) switch to astropy models for Gaussian (done)
-    2) return info about trace width (?)
-    3) add re-fit trace functionality (or break off into other method)
-    4) add other interpolation modes besides spline, maybe via
-        specutils.manipulation methods?
+        One of ``gaussian``, ``centroid``, or ``max``.
+        ``gaussian``: Fits a gaussian to the window within each bin and
+        adopts the central value as the peak. (Based on the "kosmos"
+        algorithm from James Davenport's same-named repository.)
+        ``centroid``: Takes the centroid of the window within in bin.
+        ``max``: Saves the position with the maximum flux in each bin.
+        [default: ``gaussian``]
+
+    Possible Improvements
+    ---------------------
+    1) return info about trace width (?)
+    2) add re-fit trace functionality (or break off into other method)
     """
-    bins: int = 20
+    bins: int = 4
```

Comment:
might want to explicitly mention this default change in the change log?  We're not too strict about breaking changes yet, so I don't think we need to deprecate first, but we should at least try to make the change as visible as possible.

Reply:
I can't imagine real data would likely have less than 20 dispersion pixels, but I guess our other option would be to have it based on the size of the input image (something like `max(20, npixels/4)`)

Reply:
Patrick wants the default equal to the number of dispersion columns. What do you think?

Reply:
Works for me.  We might want some logic to skip binning entirely for the case where it's equal to the image size or larger (or raise an error for the larger case).